great_expectations/profile/json_schema_profiler.py

```diff
@@ -150,14 +195,39 @@ def _create_range_expectation(
         self, key: str, details: dict
     ) -> Optional[ExpectationConfiguration]:
         """https://json-schema.org/understanding-json-schema/reference/numeric.html#range"""
-        type_ = details.get("type", None)
-        if type_ not in [JsonSchemaTypes.INTEGER.value, JsonSchemaTypes.NUMBER.value]:
+        object_types = self._get_object_types(details=details)
+        object_types = filter(
+            lambda object_type: object_type != JsonSchemaTypes.NULL.value, object_types
+        )
+        range_types = [JsonSchemaTypes.INTEGER.value, JsonSchemaTypes.NUMBER.value]
+
+        if set(object_types).issubset(set(range_types)) is False:
             return None
 
-        minimum = details.get("minimum", None)
-        maximum = details.get("maximum", None)
-        exclusive_minimum = details.get("exclusiveMinimum", None)
-        exclusive_maximum = details.get("exclusiveMaximum", None)
+        type_ = details.get("type", None)
+        any_of = details.get("anyOf", None)
+
+        minimum = None
+        maximum = None
+        exclusive_minimum = None
+        exclusive_maximum = None
+
+        if type_:
+            minimum = details.get("minimum", None)
+            maximum = details.get("maximum", None)
+            exclusive_minimum = details.get("exclusiveMinimum", None)
+            exclusive_maximum = details.get("exclusiveMaximum", None)
+        elif any_of:
+            for item in any_of:
+                item_type = item.get("type", None)
+                if item_type in range_types:
+                    minimum = item.get("minimum", None)
+                    maximum = item.get("maximum", None)
+                    exclusive_minimum = item.get("exclusiveMinimum", None)
+                    exclusive_maximum = item.get("exclusiveMaximum", None)
+                    break
+        else:
+            return None
```

Comment:
Another minor guard clause readability improvement
```suggestion
        if not type_ and not any_of:
            return None

        minimum = None
        maximum = None
        exclusive_minimum = None
        exclusive_maximum = None
        
        if type_:
            minimum = details.get("minimum", None)
            maximum = details.get("maximum", None)
            exclusive_minimum = details.get("exclusiveMinimum", None)
            exclusive_maximum = details.get("exclusiveMaximum", None)
        elif any_of:
            for item in any_of:
                item_type = item.get("type", None)
                if item_type in range_types:
                    minimum = item.get("minimum", None)
                    maximum = item.get("maximum", None)
                    exclusive_minimum = item.get("exclusiveMinimum", None)
                    exclusive_maximum = item.get("exclusiveMaximum", None)
                    break
```

Reply:
Sorry - this diff looks atrocious but it's really just moving the guard clause up top

Reply:
Good point, that is a the right way to guard and exit early. Change made.